siac/cmds.go

```diff
@@ -47,20 +48,64 @@ func updatecmd(cmd *cobra.Command, args []string) {
 		cmd.Usage()
 		return
 	}
-	fmt.Println(getResponse("/update", &url.Values{
-		"action": {args[0]},
-	}))
+	switch args[0] {
+	case "check":
+		if m, err := getResponse("/update/check", nil); err == nil {
+			if m["Available"].(bool) {
+				fmt.Printf("Update %s is available!\nRun 'siac update %s' to install it.\n", m["Version"], m["Version"])
+			} else {
+				fmt.Println("Up to date!")
+			}
+		} else {
+			fmt.Println("Update check failed:", err)
+		}
+
+	default:
+		_, err := getResponse("/update/apply", &url.Values{
+			"version": {args[0]},
+		})
+		if err == nil {
+			fmt.Printf("Update %s applied!\nYou must restart siad for changes to take effect.\n", args[0])
+		} else {
+			fmt.Println("Couldn't apply update:", err)
+		}
+	}
 }
 
 func minecmd(cmd *cobra.Command, args []string) {
-	if len(args) != 1 {
+	if len(args) < 1 {
 		cmd.Usage()
 		return
 	}
-	// TODO: need start/stop
-	fmt.Println(getResponse("/mine", &url.Values{
-		"toggle": {args[0]},
-	}))
+	switch args[0] {
+	case "start":
+		_, err := getResponse("/miner/start", &url.Values{
+			"threads": {args[1]},
+		})
+		if err != nil {
+			fmt.Println("Could not start miner:", err)
+		} else {
+			fmt.Println("Mining on " + args[1] + " threads.")
+		}
+	case "stop":
+		_, err := getResponse("/miner/stop", nil)
+		if err != nil {
+			fmt.Println("Could not stop miner:", err)
+		} else {
+			fmt.Println("Stopped mining.")
+		}
+
+	case "status":
+		m, err := getResponse("/miner/status", nil)
+		if err != nil {
+			fmt.Println("Could not get miner status:", err)
+		} else {
```

Comment:
just going back to the first one, I think it would be better to:

```go
if err != nil {
    fmt.Println()
    continue
}
fmt.Println()
```
as opposed to what you have because most of the time when you error check you throw in a control statement (ie `return`), so it's good to just make that a standard.